### PR TITLE
PV-2060 submission metrics queries execution in parallel

### PR DESCRIPTION
**Pull request by hardik-desai-kcl**

### CODE REVIEW

File: src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsQueryDataContext.cs [hunk 1]
```diff
@@ -193,40 +183,35 @@ public async Task<DataLockTypeCounts> GetDataLockCounts(long ukprn, long jobId,
...
-                .ToListAsync(cancellationToken);
-        }
-    }
+			return await DataLockEventNonPayablePeriods
+				.Where(period => period.Amount != 0 && latestSuccessfulJobIds.Contains(period.DataLockEvent.JobId))
+				.GroupBy(x => x.DataLockEvent.Ukprn)
+				.Select(x => new ProviderTotal() { Ukprn = x.Key, TotalAmount = x.Sum(period => period.Amount) })
+				.ToListAsync(cancellationToken);
+		}
+	}
```
Comment (tom-gough) [inline src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsQueryDataContext.cs:216]: There seem to be a lot of formatting changes here but as far as I can see no real logic changes, can you confirm this/explain any logic changes you have made?

File: src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsQueryDataContext.cs [hunk 2]
```diff
@@ -14,67 +13,58 @@
...
-        public virtual DbQuery<ProviderTotal> AlreadyPaidDataLockProviderTotals { get; set; }
-        public virtual DbSet<LatestSuccessfulJobModel> LatestSuccessfulJobs { get; set; }
-        public virtual DbSet<EarningEventModel> EarningEvent { get; protected set; }
-        public virtual DbSet<EarningEventPeriodModel> EarningEventPeriods { get; protected set; }
-        public virtual DbSet<DataLockEventModel> DataLockEvent { get; set; }
-        public virtual DbSet<DataLockEventNonPayablePeriodModel> DataLockEventNonPayablePeriods { get; set; }
-        public virtual DbSet<RequiredPaymentEventModel> RequiredPaymentEvents { get; set; }
-        public virtual DbSet<PaymentModel> Payments { get; set; }
-
-        public MetricsQueryDataContext(string connectionString)
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsQueryDataContext.cs:50]: removed string connection string constructor and added DbContextOptions contextOptions

File: src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsQueryDataContext.cs [hunk 3]
```diff
@@ -14,67 +13,58 @@
...
-            modelBuilder.ApplyConfiguration(new EarningEventPeriodModelConfiguration());
-            modelBuilder.ApplyConfiguration(new DataLockEventModelConfiguration());
-            modelBuilder.ApplyConfiguration(new DataLockEventNonPayablePeriodModelConfiguration());
-            modelBuilder.ApplyConfiguration(new DataLockEventNonPayablePeriodFailureModelConfiguration());
-            modelBuilder.ApplyConfiguration(new RequiredPaymentEventModelConfiguration());
-            modelBuilder.ApplyConfiguration(new PaymentModelConfiguration());
-            modelBuilder.ApplyConfiguration(new LatestSuccessfulJobModelConfiguration());
-        }
-
-        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsQueryDataContext.cs:69]: removed this as its not required with DbContextOptions constructor and its done at Registration time

File: src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsQueryDataContext.cs [hunk 4]
```diff
@@ -193,40 +183,35 @@ public async Task<DataLockTypeCounts> GetDataLockCounts(long ukprn, long jobId,
...
-                DataLock7 = dataLockCounts.FirstOrDefault(amount => amount.DataLockType == (byte)DataLockErrorCode.DLOCK_07)?.Count ?? 0,
-                DataLock8 = dataLockCounts.FirstOrDefault(amount => amount.DataLockType == (byte)DataLockErrorCode.DLOCK_08)?.Count ?? 0,
-                DataLock9 = dataLockCounts.FirstOrDefault(amount => amount.DataLockType == (byte)DataLockErrorCode.DLOCK_09)?.Count ?? 0,
-                DataLock10 = dataLockCounts.FirstOrDefault(amount => amount.DataLockType == (byte)DataLockErrorCode.DLOCK_10)?.Count ?? 0,
-                DataLock11 = dataLockCounts.FirstOrDefault(amount => amount.DataLockType == (byte)DataLockErrorCode.DLOCK_11)?.Count ?? 0,
-                DataLock12 = dataLockCounts.FirstOrDefault(amount => amount.DataLockType == (byte)DataLockErrorCode.DLOCK_12)?.Count ?? 0,
-            };
-        }
-
-        public void SetTimeout(TimeSpan timeout)
```
Comment (hardik-desai-kcl) [inline src/SFA.DAS.Payments.Monitoring.Metrics.Data/MetricsQueryDataContext.cs:215]: removed this as its not required with DbContextOptions constructor and its done at Registration time

**Review by tom-gough — APPROVED** [review]

**Pull request merged**